Source/WebCore/html/HTMLFormElement.cpp

```diff
@@ -374,7 +375,7 @@ void HTMLFormElement::submit(Event* event, bool processingUserGesture, FormSubmi
     if (!view || !frame)
         return;
 
-    if (m_isSubmittingOrPreparingForSubmission) {
+    if (trigger != SubmittedByJavaScript && m_isSubmittingOrPreparingForSubmission) {
         m_shouldSubmit = true;
```

Comment:
We should probably refactor this code so that m_shouldSubmit is a local variable in submitIfPossible instead of an instance variable like this in a follow up.

Reply:
@AlexsanderDamaceno : I think that whole design is wrong but that's a topic for another issue / PR.